text/0000-manifest-list-buildpacks.md

```diff
@@ -0,0 +1,133 @@
+# Publish multi-arch buildpacks
+
+## Summary
+
+In order to enable end users to create arm64 images with paketo buildpacks I am proposing that paketo start publishing multi-arch buildpack images for all buildpacks. @dmikusa put together [this](https://github.com/dmikusa/paketo-arm64) excellent guide for building arm64 containers with paketo buildpacks, but it puts a heavy burden on end users to create and maintain copies of paketo buildpacks. This proposal is to start publishing manifest list images that support x86_64 and arm64 architectures. This would be accomplished by using `docker manifest` or similar.
+
+For simplicity I will refer to x86_64 as amd64 throughout this document.
+
+## Motivation
+
+The number one question I get about [paketo] buildpacks is whether they support arm. It is safe to say that workloads are generally moving to arm processors and end users are already evaluating how to build containers on arm. It is difficult for me as an end user to seriously consider building arm images with paketo buildpacks without putting in some engineering effort into building arm versions of the paketo buildpacks I may need. While the above example is definitely helpful, a lot of the steps can be built into the github actions workflows that create and release buildpacks for general consumption. I believe arm support was the number one requested roadmap item for 2023, and this proposal is an attempt to jump start that process.
+
+
+## Detailed Explanation
+
+Right now binaries are created on/for amd64 and then packaged up into buildpacks using pack. It should be easy to create binaries for arm64 which can then be packaged up into buildpacks using pack. These architecture-specific buildpacks can be pushed to a registry with a platform specific tag as shown in the examples below. 
+
+* `paketobuildpacks/ca-certificates:3.6-amd64`
+* `paketobuildpacks/ca-certificates:3.6-arm64`
+
+The last step would be to create a manifest list image using `docker manifest` commands (or similar) and push it to the registry.
+
+Users would pull the manifest list image such as: `paketobuildpacks/ca-certificates:3.6`
```

Comment:
I don't know much about multi-arch images, but I assume hosting a multi-arch manifest at this location is backwards-compatible with the (amd64-only) image that is currently hosted there?

Reply:
I think so but that would be a good experiment. Just to confirm.

Reply:
Yes. This image for the current platform will be pulled.

I set up the following action to test a multi-arch builders on amd64 and arm64 and it pulls the correct image. The same would be true for buildpacks.

https://github.com/jericop/cnb-builder/blob/01a194b802311e09c9acffb53470a374f8d618b3/.github/workflows/build-and-publish.yml#L90
https://github.com/jericop/cnb-builder/blob/01a194b802311e09c9acffb53470a374f8d618b3/.github/workflows/build-and-publish.yml#L102
https://github.com/jericop/cnb-builder/blob/01a194b802311e09c9acffb53470a374f8d618b3/.github/actions/multi-arch-builder-test/action.yml

Reply:
Great - thanks for clarifying.